database/query/query_text_search_test.go

```diff
@@ -1185,3 +1186,348 @@ func TestPrepareSearchContentRemoveEmojis(t *testing.T) {
 		})
 	}
 }
+
+func TestSearchExtensions_StartsWithAndContains(t *testing.T) {
+	t.Parallel()
+	db := helperNewDatabase(t)
+	defer db.Close()
+	profiles := []struct {
+		name        string
+		displayName string
+		masterKey   string
+		privateKey  string
+	}{
+		{"alice", "Alice Smith", "", ""},
+		{"alison", "Alison Cooper", "", ""},
+		{"alexander", "Alexander Great", "", ""},
+		{"bob", "Bob Jones", "", ""},
+		{"charlie", "Charlie Brown", "", ""},
+	}
+
+	for i := range profiles {
+		profiles[i].privateKey, profiles[i].masterKey = model.GenerateKeyPair()
+
+		content := `{"name":"` + profiles[i].name + `","display_name":"` + profiles[i].displayName + `"}`
+		profileEvent := &model.Event{
+			Event: nostr.Event{
+				Kind:      nostr.KindProfileMetadata,
+				CreatedAt: nostr.Now(),
+				Content:   content,
+			},
+		}
+		require.NoError(t, profileEvent.SignWithAlg(profiles[i].privateKey, model.SignAlgEDDSA, model.KeyAlgCurve25519))
+		require.NoError(t, db.AcceptEvents(t.Context(), profileEvent))
+	}
+
+	t.Run("StartsWith search - find alice, alison, alexander", func(t *testing.T) {
+		stored := helperSelectEvents(t, db, model.Filter{
+			Kinds:  []int{nostr.KindProfileMetadata},
+			Search: `startsWith "ali"`,
+		})
+		require.Equal(t, len(stored), 2)
+		names := make(map[string]bool)
+		for _, ev := range stored {
+			var content struct {
+				Name string `json:"name"`
+			}
+			require.NoError(t, json.Unmarshal([]byte(ev.Content), &content))
+			names[content.Name] = true
+		}
+		require.True(t, names["alice"] || names["alison"], "Should find alice/alison")
+	})
+	t.Run("StartsWith search - find only bob", func(t *testing.T) {
+		stored := helperSelectEvents(t, db, model.Filter{
+			Kinds:  []int{nostr.KindProfileMetadata},
+			Search: `startsWith "bob"`,
+		})
+		require.Equal(t, len(stored), 1)
+		foundBob := false
+		for _, ev := range stored {
+			var content struct {
+				Name string `json:"name"`
+			}
+			require.NoError(t, json.Unmarshal([]byte(ev.Content), &content))
+			if content.Name == "bob" {
+				foundBob = true
+				break
+			}
+		}
+		require.True(t, foundBob, "Should find bob in search results")
+	})
+
+	t.Run("Contains search - find 'ali' anywhere", func(t *testing.T) {
+		stored := helperSelectEvents(t, db, model.Filter{
+			Kinds:  []int{nostr.KindProfileMetadata},
+			Search: `contains "ali"`,
```

Comment:
Isn't it basically startsWith as both alice and alison starts with ali? Be might need one more sample